### ISSUE TITLE
unrecognized control parameter RESOURCESTRATEGY

### ISSUE THREAD

**Issue by NLaws** [jump-dev/Xpress.jl#130]
Though `RESOURCESTRATEGY` is in the list from `julia -e "using Xpress; println(keys(Xpress.XPRS_ATTRIBUTES))"`, when I try to use `RESOURCESTRATEGY` I get:
```julia
ERROR: Unrecognized control parameter: 8297.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] setcontrol!(::Xpress.XpressProblem, ::Int32, ::Int64) at /Users/nlaws/.julia/packages/Xpress/ab5Cg/src/helper.jl:124
 [3] setcontrol!(::Xpress.XpressProblem, ::Int64, ::Int64) at /Users/nlaws/.julia/packages/Xpress/ab5Cg/src/helper.jl:110
 [4] set(::Xpress.Optimizer, ::MathOptInterface.RawParameter, ::Int64) at /Users/nlaws/.julia/packages/Xpress/ab5Cg/src/MOI/MOI_wrapper.jl:527
 [5] empty!(::Xpress.Optimizer) at /Users/nlaws/.julia/packages/Xpress/ab5Cg/src/MOI/MOI_wrapper.jl:284
 [6] Xpress.Optimizer(; kwargs::Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol},NamedTuple{(:RESOURCESTRATEGY,),Tuple{Int64}}}) at /Users/nlaws/.julia/packages/Xpress/ab5Cg/src/MOI/MOI_wrapper.jl:271
 [7] #7 at ./REPL[6]:1 [inlined]
 [8] _instantiate_and_check(::var"#7#8") at /Users/nlaws/.julia/packages/MathOptInterface/ZJFKw/src/instantiate.jl:56
 [9] instantiate(::Function; with_bridge_type::Type{Float64}, with_names::Bool) at /Users/nlaws/.julia/packages/MathOptInterface/ZJFKw/src/instantiate.jl:107
 [10] set_optimizer(::Model, ::Function; bridge_constraints::Bool) at /Users/nlaws/.julia/packages/JuMP/qhoVb/src/optimizer_interface.jl:66
 [11] Model(::Function; bridge_constraints::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/nlaws/.julia/packages/JuMP/qhoVb/src/JuMP.jl:235
 [12] Model(::Function) at /Users/nlaws/.julia/packages/JuMP/qhoVb/src/JuMP.jl:234
 [13] top-level scope at REPL[6]:1
```
Steps to recreate:
```julia
using Xpress, Jump
m = Model(()->Xpress.Optimizer(RESOURCESTRATEGY=1))
```

**Comment by joaquimg** [comment]
It might be a very recent parameter.
You can add manually in: https://github.com/jump-dev/Xpress.jl/blob/master/src/common.jl
(PR's accepted)
You need the number from the header file.
If you put that parameter in the correct order on common.jl it will be better for future maintenance.

**Comment by NLaws** [comment]
> You can add manually in: https://github.com/jump-dev/Xpress.jl/blob/master/src/common.jl

@joaquimg  `RESOURCESTRATEGY` is already in common.jl:
https://github.com/jump-dev/Xpress.jl/blob/2bdc9b796aeccc130102ed7caaa37668e01a9c9d/src/common.jl#L248

> You need the number from the header file.

What is the "header file"? Maybe the number is wrong?

**Comment by NLaws** [comment]
Fixed in https://github.com/jump-dev/Xpress.jl/pull/132 (not merged yet). I also found two more missing control parameters, `MAXMEMORYSOFT` and `MAXMEMORYHARD`.

**Comment by joaquimg** [comment]
> What is the "header file"?

On windows, you would go to xpressmp folder and then `include` and there will be a `xprs.h` (`h` as in header for C).
This file includes all the numbers if needed. It seems only `helper.jl` was missing stuff though.

**Comment by kdheepak** [comment]
This script might be old, but maybe we should run it again and see what has changed:

https://github.com/jump-dev/Xpress.jl/blob/master/scripts/generate.jl

**Comment by joaquimg** [comment]
@kdheepak the problem is helper.jl not being automatically generated.
That is why changes are only needed there.

**Comment by joaquimg** [comment]
closed by #132
